search/retrieval.cpp

```diff
@@ -226,12 +230,46 @@ unique_ptr<coding::CompressedBitVector> RetrieveAddressFeaturesImpl(
   vector<uint64_t> features;
   FeaturesCollector collector(cancellable, features);
 
-  WithSearchTrieRoot<TValue>(context.m_value, [&](TrieRoot<TValue> const & root) {
-    MatchFeaturesInTrie(
-        params, root,
-        [&holder](uint32_t featureIndex) { return !holder.ModifiedOrDeleted(featureIndex); },
-        collector);
-  });
+  // TODO (@y): this code is highly depends on the fact that the
+  // function is called on a single-token query params. In any case,
+  // this code must be fixed ASAP, as this is the wrong place for DFA
+  // creation.
+  ASSERT_EQUAL(1, params.GetNumTokens(), ());
+
+  for (size_t i = 0; i < params.GetNumTokens(); ++i)
+  {
+    if (params.IsPrefixToken(i))
+    {
+      using DFA = PrefixDFAModifier<UniStringDFA>;
+
+      SearchTrieRequest<DFA> request;
```

Comment:
Actually it's not easy to factor out the common code. See the comment above - I'm planning to get rid of this code ASAP.